src/webglimpse/timeline/timeline_events_row.ts

```diff
@@ -705,39 +710,87 @@ module Webglimpse {
                 '                                                                ',
                 '  attribute vec2 a_XyFrac;                                      ',
                 '  attribute vec4 a_Color;                                       ',
-                '                                                                ',
+                '  attribute vec2 a_relativeXy;                                   ',
+                '  attribute float a_fillPattern;                                    ',     
+                '                                                                ',                
                 '  varying vec4 v_Color;                                         ',
+                '  varying vec2 v_relativeXy;                                     ', 
+                '  varying float v_fillPattern;                                     ',                                               
                 '                                                                ',
                 '  void main( ) {                                                ',
                 '      gl_Position = vec4( ( -1.0 + 2.0*a_XyFrac ), 0.0, 1.0 );  ',
                 '      v_Color = a_Color;                                        ',
+                '      v_relativeXy = a_relativeXy;                                ',
+                '      v_fillPattern = a_fillPattern;                                ',                
                 '  }                                                             ',
                 '                                                                '
             );
 
-        var program = new Program( xyFrac_vColor_VERTSHADER, varyingColor_FRAGSHADER );
+
+         var fillPattern_FRAGSHADER = Webglimpse.concatLines(
+                '                                                                           ',
+				'  precision highp float;                                                   ',
+				'  varying vec4 v_Color;                                                    ',
+   				'  varying vec2 v_relativeXy;                                                ',
+   				'  varying float v_fillPattern;                                              ',                   
```

Comment:
Question about this. My impression was that the geometry for all of the events in a row are sent to the fragment shader and would need to know for each quad whether or not it has a fill pattern or not and the color for each event.  Wouldn't it have to be varying? How would you know which event/quad you are in since the fill pattern/color are specific to each event?